compiler/rustc_data_structures/src/fingerprint.rs

```diff
@@ -13,28 +13,23 @@ impl Fingerprint {
     pub const ZERO: Fingerprint = Fingerprint(0, 0);
 
     #[inline]
-    pub fn new(_0: u64, _1: u64) -> Fingerprint {
-        Fingerprint(_0, _1)
-    }
-
-    #[inline]
-    pub fn from_smaller_hash(hash: u64) -> Fingerprint {
-        Fingerprint(hash, hash)
+    pub fn new(_0: Hash64, _1: Hash64) -> Fingerprint {
+        Fingerprint(_0.as_u64(), _1.as_u64())
```

Comment:
Why not directly store 2 Hash64? Or directly a Hash128?

Reply:
I'm _trying_ to do some encapsulation here (it's not really working, probably needs follow-up PRs to improve it).

Users of `Fingerprint` store separate 64-bit hashes in it and there is also this user which creates a `Fingerprint` which is only half hash: https://github.com/rust-lang/rust/blob/fd57c6b407d5f038060d7b94ce9af22ea4b0dcf0/compiler/rustc_middle/src/dep_graph/dep_node.rs#L357-L361

After looking through this a lot I think the `Fingerprint` type should probably be overhauled or discarded. Different users of `Fingerprint` seem to be getting very different things out of it which seem mostly conceptually disjoint.